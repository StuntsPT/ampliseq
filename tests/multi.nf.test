nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    profile "test_multi"

    test("-profile test_multi") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                /* New syntax, is not satisfied
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_ampliseq_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
                */
                // Old syntax, provisionally used
                { assert new File("$outputDir/pipeline_info/software_versions.yml").exists() },
                { assert snapshot(path("$outputDir/overall_summary.tsv")).match("overall_summary_tsv") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                path("$outputDir/barrnap/rrna.bac.gff"),
                                path("$outputDir/barrnap/rrna.euk.gff"),
                                path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                path("$outputDir/dada2/ASV_table.tsv"),
                                path("$outputDir/dada2/ref_taxonomy.rdp_18.txt"),
                                path("$outputDir/dada2/DADA2_stats.tsv"),
                                path("$outputDir/dada2/DADA2_table.rds"),
                                path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert new File("$outputDir/dada2/ASV_tax.rdp_18.tsv").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1a_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1a_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2a_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2a_2_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/input/Samplesheet_multi.tsv")).match("input") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt")).match("multiqc") },
                { assert new File("$outputDir/qiime2/abundance_tables/abs-abund-table-2.tsv").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/abs-abund-table-3.tsv").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/abs-abund-table-4.tsv").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/abs-abund-table-5.tsv").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/abs-abund-table-6.tsv").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/count_table_filter_stats.tsv").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/filtered-table.qza").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/feature-table.biom").exists() },
                { assert new File("$outputDir/qiime2/abundance_tables/feature-table.tsv").exists() },
                { assert new File("$outputDir/qiime2/input/rep-seqs.qza").exists() },
                { assert new File("$outputDir/qiime2/input/table.qza").exists() },
                { assert new File("$outputDir/qiime2/input/taxonomy.qza").exists() },
                { assert new File("$outputDir/qiime2/rel_abundance_tables/rel-table-2.tsv").exists() },
                { assert new File("$outputDir/qiime2/rel_abundance_tables/rel-table-3.tsv").exists() },
                { assert new File("$outputDir/qiime2/rel_abundance_tables/rel-table-4.tsv").exists() },
                { assert new File("$outputDir/qiime2/rel_abundance_tables/rel-table-5.tsv").exists() },
                { assert new File("$outputDir/qiime2/rel_abundance_tables/rel-table-6.tsv").exists() },
                { assert new File("$outputDir/qiime2/rel_abundance_tables/rel-table-ASV.tsv").exists() },
                { assert new File("$outputDir/qiime2/rel_abundance_tables/rel-table-ASV_with-DADA2-tax.tsv").exists() },
                { assert new File("$outputDir/qiime2/representative_sequences/filtered-sequences.qza").exists() },
                { assert new File("$outputDir/qiime2/representative_sequences/rep-seq.fasta").exists() },
                { assert snapshot(path("$outputDir/qiime2/representative_sequences/descriptive_stats.tsv"),
                                path("$outputDir/qiime2/representative_sequences/seven_number_summary.tsv")).match("qiime2") },
                { assert new File("$outputDir/summary_report/summary_report.html").exists() }
            )
        }
    }
}
