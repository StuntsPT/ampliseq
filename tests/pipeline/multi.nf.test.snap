{
    "input": {
        "content": [
            "Samplesheet_multi.tsv:md5,25b97db179c42b568e28f00a1c475f78"
        ],
        "timestamp": "2023-05-28T21:15:03+0000"
    },
    "qiime2": {
        "content": [
            "descriptive_stats.tsv:md5,f6c6527f32005a005fef4d1830874a3e",
            "seven_number_summary.tsv:md5,4b177ed9a865e3320953b69860a435ef"
        ],
        "timestamp": "2023-05-28T21:15:03+0000"
    },
    "overall_summary_tsv": {
        "content": [
            "overall_summary.tsv:md5,cc01727705136cc5d096d373237e1e06"
        ],
        "timestamp": "2023-05-28T21:15:03+0000"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,f786f2a262c3b7bf1eca475dc2c9bef6",
            "ASV_table.tsv:md5,91ca9b9ab06a873033c29af4f93dfa80",
            "ref_taxonomy.rdp_18.txt:md5,c20d8e186e2893ba7818a0115edb013e",
            "DADA2_stats.tsv:md5,1413d7b7abcdd9344efe78b550112345",
            "DADA2_table.rds:md5,e55f6c929176c853ac4d91123532df9c",
            "DADA2_table.tsv:md5,e9d7e7dfabdd1f9d3bbaa45eba545367"
        ],
        "timestamp": "2023-05-28T21:15:03+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,07229e8767a6165fdc2e392d75aec74a",
            "rrna.bac.gff:md5,5e2dc0da75181e69c61ad49fe877848a",
            "rrna.euk.gff:md5,12cd46457b8fcf5853a6cfba39dd7c63",
            "rrna.mito.gff:md5,737567148a9252d93deb529a06f3239c"
        ],
        "timestamp": "2023-05-28T21:15:03+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_general_stats.txt:md5,5cca03b6090ae25c5c3d4b5f49a7acb8",
            "multiqc_fastqc.txt:md5,7d23984b06e4251f17e1cfce7a04aaae"
        ],
        "timestamp": "2023-05-28T21:15:03+0000"
    }
}
